### DISCUSSION TITLE
IFieldUserValue.Email is not loaded with caml query

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#640] [Q&A] [answered]
I use the Getting the list, like in the documentation then I have viewXml that has `<FieldRef Name='RequestFor' />` where RequestFor is a person field in the list.
I can't figure out how to load all the properties of IFieldUserValue and 
```
         await DPCDocumentsList.LoadItemsByCamlQueryAsync(new CamlQueryOptions()
                {
                    ViewXml = viewXml,
                    DatesInUtc = true
                });
```
does not help 
when  I try: ` Console.WriteLine((listItem["RequestFor"] as IFieldUserValue).Principal);` or ` Console.WriteLine((listItem["RequestFor"] as IFieldUserValue).Email);`

How do I load it?

Thank you

**Comment by jansenbe** [comment]
@Ofer-Gal , what you see is the expected behavior as only the user id is returned. You have a number of options:

1. Switch to using `LoadListDataAsStreamAsync` instead of `LoadItemsByCamlQueryAsync` as the underlying API call providers richer data. Here's what you get back for a user field. See https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#reading-list-items for details on the 4 supported list item read methods, each have their pro's and con's.

![image](https://user-images.githubusercontent.com/7451219/142368965-93d4ca79-79c2-41c7-93c0-7689e42d5603.png)

2. Call `context.Web.EnsureUserAsync()` passing in the id value you get back from the `LoadItemsByCamlQueryAsync` call

![image](https://user-images.githubusercontent.com/7451219/142370062-f10989f3-4251-4c90-a043-b681ba029c76.png)

Here's an example of what `EnsureUserAsync` will return:

![image](https://user-images.githubusercontent.com/7451219/142370512-db6d6f6f-aca3-4c97-beb6-789a7f816711.png)

**Comment by Ofer-Gal** [comment, marked as answer]
None of the options worked
the first option shows:
`		Message	"Property Email was not yet loaded"	string
		Message	"Property Principal was not yet loaded"	string
`
![image](https://user-images.githubusercontent.com/5054931/142427411-42e8224a-3f11-44b8-9204-1525b573a713.png)

The Author does not show at all just the AuthorId.
![image](https://user-images.githubusercontent.com/5054931/142428252-311aea12-2fac-4d0c-b609-ac381d71f909.png)


using `var requester = await pnpContext.Web.EnsureUserAsync(listItem["RequestForId"].ToString());` throws exception:
`"SharePoint Rest service exception"`
The detail error is:
`
  at PnP.Core.Services.BatchClient.ProcessSharePointRestBatchResponseContent(Batch batch, String batchResponse)\r\n  
 at PnP.Core.Services.BatchClient.<ProcessSharePointRestBatchResponse>d__48.MoveNext()\r\n   
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
 at PnP.Core.Services.BatchClient.<ExecuteSharePointRestBatchAsync>d__46.MoveNext()\r\n   
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
 at PnP.Core.Services.BatchClient.<ExecuteBatch>d__36.MoveNext()\r\n   
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n 
`
The "was not yet loaded" is like in CSOM where do I load my "RequestFor" person field?
Added:
I got the latest and found out the exception has is because the user not found. But the message is too general
It shows under ex.Error as Message
![image](https://user-images.githubusercontent.com/5054931/142438768-ec1b9cbc-0d77-4881-9254-14d67f80b018.png)

But Writing `ex.Error`  returns `error CS1061: 'Exception' does not contain a definition for 'Error' and no accessible extension method 'Error' accepting a first argument of type 'Exception' could be found (are you missing a using directive or an assembly reference?)`
What am I missing?

Thanks

**Reply by jansenbe** [reply]
@Ofer-Gal : are you loading the needed fields in your CAML? Maybe try with leaving the `<ViewFields>...</ViewFields>` out of the CAML query.

About `EnsureUserAsync`...I was wrong, you need to use `GetUserByIdAsync`, docs are here: https://pnp.github.io/pnpcore/using-the-sdk/security-users.html#getting-users

About the exception details: just do ex.ToString() to get all the details, that's the easiest.

**Reply by Ofer-Gal** [reply]
I tried another combination with LoadListDataAsStreamAsync and now everything works fine.
Thank you much for the help.
